code/_onclick/hud/screen_objects.dm

```diff
@@ -750,3 +750,102 @@ INITIALIZE_IMMEDIATE(/atom/movable/screen/splash)
 	name = "stamina"
 	icon_state = "stamina0"
 	screen_loc = ui_stamina
+
+#define HUNGER_STATE_FAT 2
+#define HUNGER_STATE_FULL 1
+#define HUNGER_STATE_FINE 0
+#define HUNGER_STATE_HUNGRY -1
+#define HUNGER_STATE_STARVING -2
+
+/atom/movable/screen/hunger
+	name = "hunger"
+	icon_state = "hungerbar"
+	base_icon_state = "hungerbar"
+	screen_loc = ui_hunger
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	/// What state of hunger are we in?
+	VAR_PRIVATE/state = HUNGER_STATE_FINE
+	/// What food icon do we show by the bar
+	var/food_icon = 'icons/obj/food/burgerbread.dmi'
+	/// What food icon state do we show by the bar
+	var/food_icon_state = "hburger"
+	/// The image shown by the bar.
+	VAR_PRIVATE/image/food_image
```

Comment:
(really nitty nit) imo it feels like we should organize the `VAR_PRIVATE`s in one section and leave the others at the top but it's whatever.

Reply:
I organized it like this so the food icon and state would be above the food image